src/models/BaseModel.py

```diff
@@ -125,27 +129,38 @@ def preprocess_data(self):
             binary_encoded_categories,
             [pandas_vector_to_list(self.salary_feature)]
         ))
-        concatanated_features = np.transpose(concatanated_features)
-
-        feature_list = [('desc_words', len(onehot_encoded_desc_words)),
-                        ('contract_type', len(onehot_encoded_contract_type)),
-                        ('contract_time', len(onehot_encoded_contract_time)),
-                        ('company', len(binary_encoded_company)),
-                        ('source', len(binary_encoded_source)),
-                        ('town', len(binary_encoded_town)),
-                        ('region', len(binary_encoded_region)),
-                        ('job_titles', len(binary_encoded_job_titles)),
-                        ('job_modifiers', len(binary_encoded_job_modifiers)),
-                        ('categories', len(binary_encoded_categories)),
-                        ]
-        print('Pre-processing ends \n')
-        return (concatanated_features, feature_list)
+        concatenated_features = np.transpose(concatenated_features)
+
+        concatenated_column_names = np.concatenate((
+            col_names_1,
+            col_names_2,
+            col_names_3,
+            col_names_4,
+            col_names_5,
+            col_names_6,
+            col_names_7,
+            col_names_8,
+            col_names_9,
+            col_names_10,
+            ['Salary']
+        ))
+        print('Pre-processing finished \n')
+        return (concatenated_features, concatenated_column_names)
 
     @staticmethod
-    def export_data(list_to_write, file_name):
+    def export_data(list_to_write, column_names, file_name):
         print('Exporting data to ../data/' + file_name)
         print('Will read from there next time to avoid pre-processing')
         f = open('../data/' + file_name + '.csv', 'a')
+
+        # Write column names first
+        for index, item, in enumerate(column_names):
```

Comment:
```
f.write(','.join(column_names))
```
should be able to replace the entire loop here. i'm assuming you want "something,something,something,something" ~~and that each something is in it's own cell in the `column_names` list~~. may have to play around with the `str` part, worst case you can do that through a list comprehension in line.